firedrake/functionspace.py

```diff
@@ -123,8 +123,11 @@ def FunctionSpace(mesh, family, degree=None, name=None, vfamily=None,
 
     # Support foo x Real tensorproduct elements
     real_tensorproduct = False
-    if isinstance(element, ufl.TensorProductElement):
-        a, b = element.sub_elements()
+    scalar_element = element
+    if isinstance(element, ufl.VectorElement):
```

Comment:
I guess you could also allow it for `ufl.TensorElement`?

Reply:
Sure, done.